pkg/iobuf/bufferedreaderseeker.go

```diff
@@ -74,22 +77,21 @@ func NewBufferedReaderSeeker(r io.Reader) *BufferedReadSeeker {
 	const defaultThreshold = 1 << 24 // 16MB threshold for switching to file buffering
 
 	var (
-		buf      *buffer.Buffer
-		seeker   io.Seeker
-		seekable bool
+		buf        *buffer.Buffer
+		seeker     io.Seeker
+		isSeekable bool
 	)
 
-	if isSeekable(r) {
-		seeker, _ = r.(io.Seeker)
-		seekable = true
-	} else {
+	seeker = asSeeker(r)
+	if seeker == nil {
+		isSeekable = true
```

Comment:
`isSeekable` should be set to `false`, correct? But what do you think about removing `isSeekable` altogether in favor of `seeker != nil` checks?

Reply:
I initially had it like that but found the boolean a bit easier to reason about. I'll change it back.